src/components/InventoryV2/BatchSummary.tsx

```diff
@@ -1,44 +1,87 @@
+import React, { useCallback, useEffect, useState } from 'react';
 import { Grid, useTheme } from '@mui/material';
+import _ from 'lodash';
 import strings from 'src/strings';
 import { Batch } from 'src/types/Batch';
-import OverviewItemCard from '../common/OverviewItemCard';
-import { SubLocationService } from 'src/services';
-import React, { useEffect, useState } from 'react';
+import OverviewItemCard from 'src/components/common/OverviewItemCard';
 import Link from 'src/components/common/Link';
 import { APP_PATHS } from 'src/constants';
+import { useSubLocations } from 'src/components/InventoryV2/form/useSubLocations';
+import SubLocationsDropdown from 'src/components/InventoryV2/form/SubLocationsDropdown';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { requestSaveBatch } from 'src/redux/features/batches/batchesAsyncThunks';
+import { selectBatchesRequest } from 'src/redux/features/batches/batchesSelectors';
+import useSnackbar from 'src/utils/useSnackbar';
+import { BatchData } from 'src/services/NurseryBatchService';
 
 interface BatchSummaryProps {
   batch: Batch;
 }
 
 export default function BatchSummary(props: BatchSummaryProps): JSX.Element {
-  const { batch } = props;
+  const dispatch = useAppDispatch();
   const theme = useTheme();
-  const [batchSubLocations, setBatchSubLocations] = useState<string[]>([]);
+  const snackbar = useSnackbar();
+
+  const [batch, setBatch] = useState<Batch>(props.batch);
+  const { availableSubLocations, selectedSubLocations } = useSubLocations(batch.facilityId, batch);
+
+  const [requestId, setRequestId] = useState('');
+  const batchesRequest = useAppSelector(selectBatchesRequest(requestId));
+
+  const [showSubLocationEdit, setShowSubLocationEdit] = useState<boolean>(false);
+
+  const handleUpdateSubLocations = useCallback(
+    (setFn: (previousBatch: Batch) => Batch) => {
+      const nextBatch = setFn(batch);
+      setBatch(nextBatch);
+    },
+    [batch]
+  );
+
+  const toggleSubLocationEdit = useCallback(() => {
+    const nextShowSubLocationEdit = !showSubLocationEdit;
+
+    // If we're "turning off" the edit mode, and the sub locations aren't the same, we need to update the batch
+    if (!nextShowSubLocationEdit && !_.isEqual(props.batch.subLocationIds, batch.subLocationIds)) {
+      const request = dispatch(requestSaveBatch({ batch }));
```

Comment:
thoughts on a bullk update/dispatch for batches that need updating?

Reply:
Not sure I follow... there is only a single batch in this view.

Reply:
My bad, I thought the sublocations was part of a species summary - and applied to a list of batches. That's the older view. !Please ignore!